core/src/test/scala/unit/kafka/utils/TestUtils.scala

```diff
@@ -475,9 +475,11 @@ object TestUtils extends Logging {
                         props: Option[Properties] = None): KafkaProducer[K, V] = {
 
     val producerProps = props.getOrElse(new Properties)
+    
     producerProps.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, brokerList)
     producerProps.put(ProducerConfig.ACKS_CONFIG, acks.toString)
-    producerProps.put(ProducerConfig.MAX_BLOCK_MS_CONFIG, maxBlockMs.toString)
+    if(!producerProps.containsKey(ProducerConfig.MAX_BLOCK_MS_CONFIG))
+       producerProps.put(ProducerConfig.MAX_BLOCK_MS_CONFIG, maxBlockMs.toString)
```

Comment:
It's not good to introduce this inconsistency here. Can you not just pass the parameter via the `createNewProducer` call?